homeassistant/components/intellifire/climate.py

```diff
@@ -0,0 +1,127 @@
+"""Intellifire Climate Entities."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.climate import (
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    ClimateEntity,
+    ClimateEntityDescription,
+    ClimateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import IntellifireDataUpdateCoordinator
+from .const import DOMAIN, LOGGER
+from .entity import IntellifireEntity
+
+
+@dataclass
+class IntellifireClimateEntityDescription(ClimateEntityDescription):
+    """Describes a fan entity."""
+
+
+INTELLIFIRE_CLIMATES: tuple[IntellifireClimateEntityDescription, ...] = (
+    IntellifireClimateEntityDescription(key="climate", name="climate"),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the fans."""
+    coordinator: IntellifireDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        IntellifireClimate(
+            coordinator=coordinator,
+            description=description,
+        )
+        for description in INTELLIFIRE_CLIMATES
+    )
+
+
+class IntellifireClimate(IntellifireEntity, ClimateEntity):
+    """Intellifire climate entity."""
+
+    entity_description: IntellifireClimateEntityDescription
+
+    _attr_hvac_modes = [HVAC_MODE_HEAT, HVAC_MODE_OFF]
+    _attr_min_temp = 0
+    _attr_max_temp = 37
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = TEMP_CELSIUS
+    last_temp = 21
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return current hvac mode."""
+        if self.coordinator.read_api.data.thermostat_on:
+            return HVAC_MODE_HEAT
+        return HVAC_MODE_OFF
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Turn on thermostat by setting a target temperature."""
+        raw_target_temp = kwargs[ATTR_TEMPERATURE]
+        self.last_temp = int(raw_target_temp)
+        LOGGER.info(
+            "Setting target temp to %sc %sf",
+            int(raw_target_temp),
+            (raw_target_temp * 9 / 5) + 32,
+        )
+        await self.coordinator.control_api.set_thermostat_c(
+            fireplace=self.coordinator.control_api.default_fireplace,
+            temp_c=self.last_temp,
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return float(self.coordinator.read_api.data.temperature_c)
+
+    @property
+    def target_temperature(self) -> float | None:
```

Comment:
```suggestion
    def target_temperature(self) -> float:
```

Can this actually be `None`?